### PR TITLE
[DEV-9966] Remove Deprecated District Filter Term

### PR DESCRIPTION
**Pull request by dporth-frb**
**Description:**
The `district` filter term was replaced with the `district-original` filter term earlier in 2023. We were previously handling both terms and sending users a deprecated message when they used the just the `district` term. This work completely removes the `district` term and any related deprecated messages.

**Technical details:**
There is no longer any conditional handling of the term `district` and unit tests have been updated to reflect the removal of this term.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. N/A - Matview impact assessment completed
5. N/A - Frontend impact assessment completed
6. [x] Data validation completed
7. N/A - Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-9966](https://federal-spending-transparency.atlassian.net/jira/software/c/projects/DEV/boards/25?modal=detail&selectedIssue=DEV-9966):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
4. Matview was not impacted by any of these changes.
5. Frontend was not impacted by any of these changes. They have already transitioned to the district-original filter term
7. No operational changes need to be done as part of this PR.
```

### CODE REVIEW

**Comment by dporth-frb** [comment]
I found some additional places that need the `district` term removed. I found these by just searching for "district" in our code base.
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L206
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L212
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L196
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L202
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L93
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L125
* https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L120

File: usaspending_api/awards/tests/unit/test_location_filter_geocode.py [hunk 1]
```diff
@@ -208,8 +208,7 @@ def test_validate_location_keys():
                     "zip": "12345",
                     "city": "Chicago",
                     "state": "IL",
-                    "county": "Yes",
```
Comment (dporth-frb) [inline usaspending_api/awards/tests/unit/test_location_filter_geocode.py:211]: Why did you remove "county"?

Comment (cthomas-frb) [reply usaspending_api/awards/tests/unit/test_location_filter_geocode.py:211]: These tests cover valid inputs and county is not valid with the district_original term. I just added a valid test that contains the county term to make sure we are still covering this scenario of having the county term present in a valid input.

**Comment by dporth-frb** [comment]
> I found some additional places that need the `district` term removed. I found these by just searching for "district" in our code base.
> 
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L206
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L212
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L196
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L202
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L93
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L125
> * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L120

@cthomas-frb did you address this comment?

**Comment by cthomas-frb** [comment]
> Member



> > I found some additional places that need the `district` term removed. I found these by just searching for "district" in our code base.
> > 
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L206
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/bulk_download/awards.md#L212
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L196
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/contracts/v2/download/awards.md#L202
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L93
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L125
> > * https://vscode.dev/github/fedspendingtransparency/usaspending-api/blob/mod/DEV-9966_Remove-deprecated-district-filter-term/usaspending_api/api_contracts/search_filters.md#L120
> 
> @cthomas-frb did you address this comment?

Yes I made those updates

**Review by collinwr — APPROVED** [review]
